src/wirecloud/commons/static/js/StyledElements/GUIBuilder.js

```diff
@@ -64,6 +55,55 @@
         return new_component;
     };
 
+    extractOptionsFromTextNode = function (element) {
+        var options = {};
+
+        for (var i = 0; i < element.childNodes.length; i++) {
```

Comment:
In order to support both options passed as JSON text, and child nodes, I changed how to get the JSON options by looking for the first non-empty text node